src/main/scala/com/lightbend/rp/sbtreactiveapp/Endpoint.scala

```diff
@@ -16,10 +16,42 @@
 
 package com.lightbend.rp.sbtreactiveapp
 
-import scala.collection.immutable.Seq
+sealed trait Endpoint {
+  def name: String
+  def port: Int
+  def protocol: String
+  def version: Option[Version]
+}
+
+case class HttpEndpoint(name: String, port: Int, ingress: Option[HttpIngress], version: Option[Version] = Some(MajorVersion)) extends Endpoint {
```

Comment:
As discussed in our standup, we'll have `ingress: Seq[HttpIngress]` for `HttpEndpoint` due to the fact that we may want to expose different HTTP path(s) based on the host.